### PR TITLE
Added reorg detection

### PR DESCRIPTION
**Pull request by zavelevsky**

### CODE REVIEW

File: src/chain-cache/ChainSync.ts [hunk 1]
```diff
@@ -268,4 +286,62 @@ export class ChainSync {
...
+  }
+
+  private async _detectReorg(currentBlock: number): Promise<boolean> {
+    logger.debug('_detectReorg called');
+    const blocksMetadata: BlockMetadata[] = this._chainCache.blocksMetadata;
+    const numberToBlockMetadata: { [key: number]: BlockMetadata } = {};
+    for (let blockMetadata of blocksMetadata) {
+      const { number, hash } = blockMetadata;
+      if (number > currentBlock) {
+        console.log(
```
Comment (lbeder) [inline src/chain-cache/ChainSync.ts:302]: logger.debug?

File: src/contracts-api/Reader.ts [hunk 2]
```diff
@@ -181,8 +197,18 @@ export default class Reader {
         id: res.id,
         token0: res.token0,
         token1: res.token1,
-        order0: res.order0,
-        order1: res.order1,
+        order0: {
```
Comment (lbeder) [inline src/contracts-api/Reader.ts:200]: Why is it necessary?

Comment (zavelevsky) [reply src/contracts-api/Reader.ts:200]: The object returning from the contract is a mix of array of values and an object - and if I simply use it as-is it eventually messes up with my serialization logic. I will tidy up the code.

Comment (lbeder) [reply src/contracts-api/Reader.ts:200]: Roger that

Comment (zavelevsky) [reply src/contracts-api/Reader.ts]: improved the code

File: src/chain-cache/ChainSync.ts (hunk 1 above)
Comment (zavelevsky) [reply src/chain-cache/ChainSync.ts]: changed to logger and added control over verbosity

**Pull request merged**